### PR TITLE
[CGData][Merger] Avoid merging the attached call target

### PR DESCRIPTION
**Pull request by kyulee-com**
For global function merging, the target of the arc-attached call must be a constant and cannot be parameterized.
This change adds a check to bypass this case in `canParameterizeCallOperand()`.

### CODE REVIEW

**Comment by kyulee-com** [comment]
cc. @nocchijiang

File: llvm/lib/CodeGen/GlobalMergeFunctions.cpp [hunk 1]
```diff
@@ -60,11 +60,17 @@ static bool canParameterizeCallOperand(const CallBase *CI, unsigned OpIdx) {
...
+  if (isCalleeOperand(CI, OpIdx)) {
     // The operand is the callee and it has already been signed. Ignore this
     // because we cannot add another ptrauth bundle to the call instruction.
-    return false;
+    if (CI->getOperandBundle(LLVMContext::OB_ptrauth).has_value())
+      return false;
+  } else {
+    // The target of the arc-attached call must be a constant and cannot be
+    // parameterized.
+    if (CI->isOperandBundleOfType(llvm::LLVMContext::OB_clang_arc_attachedcall,
```
Comment (nocchijiang) [inline llvm/lib/CodeGen/GlobalMergeFunctions.cpp:71]:
```suggestion
    if (CI->isOperandBundleOfType(LLVMContext::OB_clang_arc_attachedcall,
```

to be consistent with `CI->getOperandBundle(LLVMContext::OB_ptrauth)` above.

File: llvm/lib/CodeGen/GlobalMergeFunctions.cpp [hunk 2]
```diff
@@ -60,11 +60,17 @@ static bool canParameterizeCallOperand(const CallBase *CI, unsigned OpIdx) {
...
-  if (isCalleeOperand(CI, OpIdx) &&
-      CI->getOperandBundle(LLVMContext::OB_ptrauth).has_value()) {
+  if (isCalleeOperand(CI, OpIdx)) {
     // The operand is the callee and it has already been signed. Ignore this
     // because we cannot add another ptrauth bundle to the call instruction.
-    return false;
+    if (CI->getOperandBundle(LLVMContext::OB_ptrauth).has_value())
+      return false;
+  } else {
+    // The target of the arc-attached call must be a constant and cannot be
```
Comment (nocchijiang) [inline llvm/lib/CodeGen/GlobalMergeFunctions.cpp:69]: Is there a general method to determine whether a bundle is unsafe to optimize? If so, we can avoid enumerating all of them.

Comment (kyulee-com) [reply llvm/lib/CodeGen/GlobalMergeFunctions.cpp:69]: Not familiar with the bundle. Conservatively, we could bail out all bundle cases, but for now I bail out the known case only. I think if we run into invalid cases, the compilation will fail/crash so we can follow up.

**Comment by kyulee-com** [comment]
Any other comments? This is relatively a straightforward fix to unblock compiler crash.

**Review by nocchijiang — APPROVED** [review]
LGTM, but hope that someone else with commit access to review & approve before merging.

**Comment by kyulee-com** [comment]
@ellishg @thevinster Do you have any comments?

**Review by ellishg — APPROVED** [review]

**Pull request merged**